src/OSPSuite.Core/Domain/Services/FormulaTask.cs

```diff
@@ -322,31 +321,66 @@ public void ExpandDynamicFormulaIn(IContainer rootContainer)
                throw new CircularReferenceInSumFormulaException(dynamicFormula.Name, entityUsingFormula.Name);
 
             dynamicFormula.Criteria = updateDynamicFormulaCriteria(dynamicFormula, entityUsingFormula);
-            entityUsingFormula.Formula = dynamicFormula.ExpandUsing(allFormulaUsable, _objectPathFactory, _objectBaseFactory);
+
+            //this needs to be done after updating the criteria as dynamic tags might be added to the criteria and would 
+            //render the descriptor map list invalid
+            var allFormulaUsableDescriptor = allFormulaUsable.ToEntityDescriptorMapList();
```

Comment:
took me a while to figure this one out. We have some caching going on that was outside of the loop. In my implementation, I decided to add a dynamic tag to exclude specific children only (e.g. the direct children) of the container